### PR TITLE
Jonmv/retry 429 and 503 when feeding

### PR DESCRIPTION
**Pull request by jonmv**
@thomasht86 please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Urgh, this turned ugly, but it seems to be the only way to make it work, perhaps because of the typing—when testing a similar setup locally, the simpler form worked perfectly.

**Review by thomasht86 — APPROVED** [review]
Great. I don`t think it got _that_ ugly. 👍

**Comment by thomasht86** [comment]
Fixes #748

File: vespa/application.py [hunk 1]
```diff
@@ -1096,7 +1110,19 @@ async def query(
             json=await r.json(), status_code=r.status, url=str(r.url)
         )
 
-    @retry(wait=wait_exponential(multiplier=1), stop=stop_after_attempt(3))
+    @retry(
+        wait=wait_exponential(multiplier=1),
+        retry=retry_any(
+            retry_if_exception(lambda x: True),
```
Comment (jonmv) [inline vespa/application.py:1116]: Could consider being more selective here, later, but I don't know what errors/exceptions to expect from the HTTP library, and I couldn't find documentation of that either. We want to retry transport-related failures, but not SSL-related, and other user errors.

**Pull request merged**